pkg/pgmodel/end_to_end_tests/query_integration_test.go

```diff
@@ -570,6 +654,296 @@ func ingestQueryTestDataset(db *pgxpool.Pool, t testing.TB, metrics []prompb.Tim
 		t.Fatalf("wrong amount of metrics ingested: got %d, wanted %d", cnt, expectedCount)
 	}
 }
+func TestPromQLEndpoint(t *testing.T) {
+	if testing.Short() || !*useDocker {
+		t.Skip("skipping integration test")
+	}
+
+	promClient, err := NewPromClient(fmt.Sprintf("http://%s:%d/api/v1", testhelpers.PromHost, testhelpers.PromPort.Int()), 10*time.Second)
+	if err != nil {
+		t.Fatalf("unable to create PromQL query client for Prometheus: %s", err)
+	}
+
+	startTime := time.Unix(1577836800, 0)
+	endTime := time.Unix(1577866800, 0)
+	step := 10 * time.Second
```

Comment:
does it make sense to run the test with diff steps?

Reply:
Sure... Let me add 3-4 step sizes and update this to run for each of them.